test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -935,7 +935,10 @@ Settings closeForRestart(RestartCallback callback, int minMasterNodes) throws Ex
             }
             if (minMasterNodes >= 0) {
                 assert DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.exists(newSettings.build()) == false : "min master nodes is auto managed";
-                newSettings.put(DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.getKey(), minMasterNodes).build();
+                newSettings.put(DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.getKey(), minMasterNodes);
+                if (callbackSettings != null && INITIAL_MASTER_NODES_SETTING.exists(callbackSettings) == false) {
```

Comment:
I think returning `null` from `onNodeStopped` is a bug since it looks equivalent to returning `Settings.EMPTY`. The only place I can see this happening is `QuorumGatewayIT` in which the `onNodeStopped` override can be completely removed. Can we just do that and then assert it's not null?

Reply:
jup